src/utils/getQualityTabData.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+'use strict';
+
+import { IQualityItem } from '../interfaces/searchResponse.interface';
+import { formatDate } from './formatDate';
+import { toggleContent } from './toggleContent';
+
+const getPublicationInformation = (data: Record<string, any>[]): string => {
+  if (Array.isArray(data) && data.length > 0) {
+    const obj = data.find((item: Record<string, any>) => item?.type?.toLowerCase() === 'publication');
+    return obj?.date ? `${formatDate(obj?.date, false, true)} - Last Revision` : '';
+  }
+  return '';
+};
+
+const getLineage = (data: Record<string, any>): string => {
+  if (Object.keys(data).length && data?.default) {
+    return toggleContent(data?.default, 'lineage');
+  }
+  return '';
+};
+
+const generateConformityData = (data: Record<string, any>[]): string => {
+  if (Array.isArray(data) && data.length > 0) {
+    let tableHTML = `<table class="details-table">
+                      <thead>
+                        <tr>
+                          <th width="60%">Specification</th>
+                          <th>Degree</th>
+                          <th>Explanation</th>
+                        </tr>
+                      </thead><tbody>`;
+    data.forEach((item: Record<string, any>) => {
+      tableHTML += `<tr>
+                      <td>${item?.title}</td>
+                      <td>${item?.pass}</td>
+                      <td>${item?.explanation}</td>
```

Comment:
when these values are not available then it will show undefined. you should skip the row if all the values are not available. if any one is not available it should show empty string.

Reply:
It has now been fixed.